awaitility/src/main/java/com/jayway/awaitility/core/ConditionAwaiter.java

```diff
@@ -164,6 +165,20 @@ public void uncaughtException(Thread thread, Throwable throwable) {
         }
     }
 
+    private ExecutorService initExecutorService() {
+        return Executors.newSingleThreadExecutor(new ThreadFactory() {
+            public Thread newThread(Runnable r) {
+                Thread thread;
+                if (conditionSettings.hasAlias()) {
+                    thread = new Thread(Thread.currentThread().getThreadGroup(), r, "awaitility[" + conditionSettings.getAlias() + ']');
+                } else {
+                    thread = new Thread(Thread.currentThread().getThreadGroup(), r, "awaitility-thread");
```

Comment:
@SimY4 I meant to be able see in a log that diagnostic messages (or errors) are from different thread, so from different `await()` calls. Having the same thread name makes it tricky.

In general I don't have a good idea. `awaitility-pool-3-thread-1` doesn't look very good and it would be required to delegate somehow to the original `ThreadFactory`. Alternatively a static (and thread safe) counter could be kept.